wftests/warrior_tests/testcases/common_actions_tests/tc_verify_arith_exp.xml

```diff
@@ -0,0 +1,204 @@
+                        <Description>Verify arithmetic expression: gt</Description>
+                        <iteration_type type=""/>
+                        <Execute ExecType="Yes">
+                                <Rule Condition="" Condvalue="" Else="" Elsevalue=""/>
+                        </Execute>
+                        <context>positive</context>
+                        <impact>impact</impact>
+                        <runmode type="Standard" value=""/>
+                        <Iteration_type type="Standard"/>
+                </step>
+                <step Driver="common_driver" Keyword="verify_arith_exp" TS="8" draft="no">
+                        <Arguments>
+                                <argument name="expected" value="41"/>
+                                <argument name="comparison" value="ge"/>
+                                <argument name="expression" value="${ENV.var1}+${REPO.var2}+10"/>
+                        </Arguments>
+                        <onError action="next"/>
+                        <Description>Verify arithmetic expression: ge</Description>
+                        <iteration_type type=""/>
+                        <Execute ExecType="Yes">
+                                <Rule Condition="" Condvalue="" Else="" Elsevalue=""/>
+                        </Execute>
+                        <context>negative</context>
+                        <impact>impact</impact>
+                        <runmode type="Standard" value=""/>
+                        <Iteration_type type="Standard"/>
+                </step>
+                <step Driver="common_driver" Keyword="verify_arith_exp" TS="9" draft="no">
+                        <Arguments>
+                                <argument name="expected" value="41"/>
+                                <argument name="comparison" value="lt"/>
+                                <argument name="expression" value="${ENV.var1}+${REPO.var2}+10"/>
+                        </Arguments>
+                        <onError action="next"/>
+                        <Description>Verify arithmetic expression: lt</Description>
+                        <iteration_type type=""/>
+                        <Execute ExecType="Yes">
+                                <Rule Condition="" Condvalue="" Else="" Elsevalue=""/>
+                        </Execute>
+                        <context>positive</context>
+                        <impact>impact</impact>
+                        <runmode type="Standard" value=""/>
+                        <Iteration_type type="Standard"/>
+                </step>
+                <step Driver="common_driver" Keyword="verify_arith_exp" TS="10" draft="no">
+                        <Arguments>
+                                <argument name="expected" value="39"/>
+                                <argument name="comparison" value="le"/>
+                                <argument name="expression" value="${ENV.var1}+${REPO.var2}-1+11"/>
```

Comment:
changing ENV.var1 to ENV.var11 (simulating user typo) gives an exception 
`data_Utils.py", line 1007, in verify_arith_exp
    if not re.search(expression_pattern, expression):
File "/usr/lib/python2.7/re.py", line 142, in search
    return _compile(pattern, flags).search(string)
TypeError: expected string or buffer`

Error not gracefully handled here

Reply:
It is handled in new changes, included a test step to verify this use-case.